src/diffusers/models/vq_model.py

```diff
@@ -133,11 +136,13 @@ def encode(self, x: torch.FloatTensor, return_dict: bool = True) -> VQEncoderOut
 
     @apply_forward_hook
     def decode(
-        self, h: torch.FloatTensor, force_not_quantize: bool = False, return_dict: bool = True
+        self, h: torch.FloatTensor, force_not_quantize: bool = False, return_dict: bool = True, shape=None
     ) -> Union[DecoderOutput, torch.FloatTensor]:
         # also go through quantization layer
         if not force_not_quantize:
             quant, _, _ = self.quantize(h)
+        elif h.dtype == torch.long:
```

Comment:
```suggestion
        elif self.config.lookup_from_codebook:
```

Reply:
I don't think setting the flag in the config makes sense, but I would be ok with passing a flag to method i.e.

```
def decode(self, ..., lookup_from_codebook):
    elif lookup_from_codebook:
        quant = self.quantize.get_codebook_entry(h, shape)
```

would that be ok?

Reply:
Hmm, I disagree here because why would one ever changce `lookup_from_codebook` at runtime? MUSE's VQModel always needs to lookup from the codebook, so it's pointless to pass it everytime to the forward method. Also we should generally try to avoid passing flags to `torch.nn.Module` 's forward pass IMO.

Reply:
Ok, nw can change it